src/components/sketches/Scene4.ts

```diff
@@ -0,0 +1,170 @@
+import p5 from 'p5';
+import {
+  BUBBLE_FREQUENCY,
+  BUBBLE_FREQUENCY_ON_LINES,
+  TRANSPARENCY,
+  TRANSPARENCY_RADIUS,
+  INC,
+  INC_START,
+  MAG_INC,
+  SCL,
+  NUM_PARTICLES,
+  STROKE_WEIGHT,
+} from '../../utils/constants';
+
+export default function sketch(p: p5): void {
+  let canvas: p5.Renderer;
+
+  let start = 0;
+  let cols: number;
+  let rows: number;
+  let zoff = 0;
+  const particles: Particle[] = [];
+  let flowfield: p5.Vector[];
+  let magOff = 0;
+  const showField = false;
+
+  p.setup = () => {
+    canvas = p.createCanvas(window.innerWidth, window.innerHeight);
+    canvas.id('p5-background');
+    p.pixelDensity(1);
+    cols = p.floor(window.innerWidth / SCL);
+    rows = p.floor(window.innerHeight / SCL);
+    p.background(0);
+
+    for (let i = 0; i < NUM_PARTICLES; i++) {
+      particles[i] = new Particle();
+    }
+
+    flowfield = new Array(rows * cols);
+
+  };
+
+  p.draw = () => {
+    if (showField) {
+      p.background(0);
+    } else {
+      p.background(p.color(0, 0, 0, 5));
+    }
+    let yoff = start;
+    for (let y = 0; y < rows; y++) {
```

Comment:
i feel like this can be converted to a map but ill let you decide if you want to make it that

Reply:
maybe ill stab it another time. ik you love maps